evaluate_prediction.py

```diff
@@ -48,35 +44,43 @@ def evaluate_continuation(
     evaluate_until_onset):
     """ Given the original and the generated continuation
     of the test item (in onset/pitch dictionaries),
-    collect the following events and get tec score.
+    collect the following events and get the cardinality score.
     'onset_increment' determines the increase of onset steps in evaluation.
     For the first ioi, the last event of the prime is also required.
     'evaluate_until_onset' determines until how many quarter notes 
     after the cut-off point we evaluate.
     """
     scores = {'precision': {}, 'recall': {}, 'F1': {}}
-    no_steps = int((evaluate_until_onset - evaluate_from_onset) / onset_increment)
+    nr_steps = int((evaluate_until_onset - evaluate_from_onset)
+                   / onset_increment)
     max_onset = evaluate_until_onset + last_onset_prime
-    for step in range(no_steps+1):
+    for step in range(nr_steps + 1):
         onset = step * onset_increment + evaluate_from_onset
         cutoff = last_onset_prime + onset
         if cutoff <= max_onset:
-            original_events = [o for o in original if float(o['onset']) <= cutoff]
-            generated_events = [
-                g for g in generated if float(g['onset']) <= cutoff
-            ]
+            original_events = original[original['onset'] <= cutoff]
+            generated_events = generated[generated['onset'] <= cutoff]
```

Comment:
What type to original / generated have? I am surprised that indexing with the object itself would work. Is this some pandas syntax? I suggest adjusting the doc string to make it clear what kind of types we're dealing with.

Reply:
`original['onset'] <= cutoff` creates a boolean pandas series object. You can index with that.

Yeah, I think documenting all the functions with e.g.

```
"""Blah blah blah function description

parameters
-----------
pram_name: expected type
    description blah blah
param_name2: expected type

returns
-------
return_varname: expected type
    description blah blah
"""
```

I'll happily do that after this pr.

Reply:
Ah, now you explain it, I recognize it from R. :)
As I just wrote, I'd rather this still ended up as part of this PR, rather than make a new one. Unless you think this needs a separate review.